### ISSUE TITLE
[E] Error[2]: [optimizer.cpp::isPolymorphic::1072] Error Code 2: Internal Error (Assertion asCopyingLeafNode() ? candidateChoices.empty() : !candidateChoices.empty() failed. 

### ISSUE THREAD

**Issue by zjlovemln** [NVIDIA/TensorRT#2030]
## Description

I meet the problem when I used the trtexc tool to generate the INT8 engine with the command "./trtexec --onnx=quant_mynet_simplify.onnx --saveEngine=uant_trt_simplifier.trt  --int8"!
but it was wrong like the below Picture screenshot:

![image](https://user-images.githubusercontent.com/81769305/171991057-b932ef08-280a-4794-a84f-648fe3172ed8.png)

I used the "pytorch_quantization" tool to auto generate Q/DQ model ,and used the onnx to simplify and transform the model ,it is the onnx model structure :
![quant_mynet_simplify onnx](https://user-images.githubusercontent.com/81769305/171991302-a004a88b-6fa8-49c4-b71a-1a1d1ed3d328.png)
Looking forward to your reply, thank you!
## Environment

**TensorRT Version**: 8.2.2
**NVIDIA GPU**: NVIDIA GeForce RTX 2060
**Operating System**: Ubuntu 18.04
**PyTorch Version (if applicable)**: 1.9
**Opset version**:13

**Comment by nvpohanh** [comment]
@zjlovemln Could you try adding Q/DQ nodes between the BatchNorm and the MaxPool?

**Comment by zjlovemln** [comment]
> @zjlovemln Could you try adding Q/DQ nodes between the BatchNorm and the MaxPool?
Thank you,I am going to try that! I met another problem that I find the LSTM module can't be quantization on cuda by using quant_modules.initialize() function by "pytorch_quantization" tool, all the data only can be on the cpu! If i put the model on cuda, the problem occur: 
![image](https://user-images.githubusercontent.com/81769305/172031380-6253c887-fb0a-414e-96a5-6113de0428ec.png)
I want to know whether Q/DQ quantization supports LSTM module!

**Comment by zjlovemln** [comment]
> @zjlovemln Could you try adding Q/DQ nodes between the BatchNorm and the MaxPool?
Thank you,I am going to try that! I met another problem that I find the LSTM module can't be quantization on cuda by using quant_modules.initialize() function by "pytorch_quantization" tool, all the data only can be on the cpu! If i put the model on cuda, the problem occur:
![image](https://user-images.githubusercontent.com/81769305/172031623-29097e9c-6a6c-4ada-b256-c450587967d4.png)
I want to know whether Q/DQ quantization supports LSTM module!

**Comment by nvpohanh** [comment]
I am still checking internally.

**Comment by zjlovemln** [comment]
I checked  the function definition of torch._ cudnn_ rnn_ flatten_ weight() ,it has a variable "proj_size:_int";But In the quant_rnn.py torch._ cudnn_ rnn_ flatten_ weight() doesn't have  "proj_size " variable , so I modified the a file myself with adding  "proj_size = 0" ,as follow:
![image](https://user-images.githubusercontent.com/81769305/173782803-435c8723-4038-4508-aa14-d810fd64ea61.png)
Although the above problems are solved, new problems arise when the model is converted to onnx：
![image](https://user-images.githubusercontent.com/81769305/173783305-b5d8d2db-9c06-43f1-b62a-5969825d7708.png)
It will always be stuck in "hidden = list (ZIP (*hidden))". I hope you can help me solve this problem. I want to know that the LSTM module supports Q/DQ and ONNX model conversion. At present, I think there is a problem with the official code. I hope you can check it!

**Comment by nvpohanh** [comment]
@zjlovemln To add Q/DQ for LSTM, you can check out the Quant LSTM/RNN in the torch quant toolkit: https://github.com/NVIDIA/TensorRT/blob/main/tools/pytorch-quantization/tests/quant_rnn_test.py

@ttyio for visibility

**Comment by nvpohanh** [comment]
Closing due to >14 days without activity. Please feel free to reopen if the issue still exists. Thanks
